js/chat_app/src/redux/auth/operation.js

```diff
@@ -0,0 +1,12 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import login from '../../services/services';
+
+const loginAuth = createAsyncThunk('auth/login', async ({ username, password }) => {
+  const response = await login({ username, password });
+  const data = await response.json();
+  localStorage.setItem('auth', JSON.stringify(data));
```

Comment:
i dont think that you need all user data.

Reply:
I changed this `localStorage.setItem('auth', JSON.stringify(data));` to this `localStorage.setItem('auth', JSON.stringify({authToken: data.authToken, user: {username: data.user.username}}));`